op-node/node/config_persistence.go

```
@@ -0,0 +1,134 @@
+func (p *ActiveConfigPersistence) persist(sequencerStarted bool) error {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	data, err := json.Marshal(persistedState{SequencerStarted: &sequencerStarted})
+	if err != nil {
+		return fmt.Errorf("marshall new config: %w", err)
+	}
+	dir := filepath.Dir(p.file)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("create config dir (%v): %w", p.file, err)
+	}
+	// Write the new content to a temp file first, then rename into place
+	// Avoids corrupting the content if the disk is full or there are IO errors
+	tmpFile := p.file + ".tmp"
+	file, err := os.OpenFile(tmpFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return fmt.Errorf("open file (%v) for writing: %w", tmpFile, err)
+	}
+	defer file.Close()
+	_, err = file.Write(data)
+	if err != nil {
+		return fmt.Errorf("write new config to temp file (%v): %w", tmpFile, err)
+	}
+	if err := file.Sync(); err != nil {
+		return fmt.Errorf("sync new config temp file (%v): %w", tmpFile, err)
+	}
+
+	// Rename to replace the previous file
+	if err := os.Rename(tmpFile, p.file); err != nil {
+		return fmt.Errorf("rename temp config file to final destination: %w", err)
+	}
+	return nil
+}
+
+func (p *ActiveConfigPersistence) SequencerState() (RunningState, error) {
+	config, err := p.read()
+	if err != nil {
+		return StateUnset, err
+	}
+
+	if config.SequencerStarted == nil {
+		return StateUnset, nil
+	} else if *config.SequencerStarted {
+		return StateStarted, nil
+	} else {
+		return StateStopped, nil
+	}
+}
+
+func (p *ActiveConfigPersistence) read() (persistedState, error) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	data, err := os.ReadFile(p.file)
+	if errors.Is(err, os.ErrNotExist) {
+		return persistedState{}, nil
+	} else if err != nil {
+		return persistedState{}, fmt.Errorf("read config file (%v): %w", p.file, err)
+	}
+	var config persistedState
+	if err = json.Unmarshal(data, &config); err != nil {
```

If we want this to be extra strict:
- With `dec := json.NewDecoder()` + `dec.DisallowUnknownFields()`, we can specify it to error in case of unknown keys in the read json file.
- We could also return an error if it loads a json value with a nil `SequencerStarted` field.

---

We may need to loosen up a bit again if we wind up persisting other values to this file but makes sense to start really strict.